src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/UseNameofInPlaceOfString.cs

```diff
@@ -90,66 +88,49 @@ private void AnalyzeArgument(OperationAnalysisContext context)
             }
         }
 
-        private static IEnumerable<string> GetPropertiesInScope(OperationAnalysisContext context)
+        private static bool HasAPropertyMatchInScope(OperationAnalysisContext context, string stringText)
         {
             var containingType = context.ContainingSymbol.ContainingType;
-            // look for all of the properties in the containing type and return the property names
-            if (containingType != null)
-            {
-                foreach (var property in containingType.GetMembers().OfType<IPropertySymbol>())
-                {
-                    yield return property.Name;
-                }
-            }
+            return containingType?.GetMembers(stringText).Any(m => m.Kind == SymbolKind.Property) == true;
         }
 
-        internal static IEnumerable<string> GetParametersInScope(OperationAnalysisContext context)
+        private static bool HasAParameterMatchInScope(OperationAnalysisContext context, string stringText)
```

Comment:
Same comment here:

[suggestion] Maybe `HasAnyParameterMatchInScope` or `HasSomeParameterMatchInScope` to make it easier to read (the `AP` makes it weird to the eyes).